### DISCUSSION TITLE
Alarm History

### DISCUSSION THREAD

**Discussion by akashkumarmandal** [dmroeder/pylogix#187] [General]
Can we read alarm history of contrologix5572 PLC And using factory talk view 12.0.0 using python?

**Comment by dmroeder** [comment]
@akashkumarmandal the alarm history is stored in a database in the PC/PanelView running the FactoryTalk project, unless this is some custom alarm history written in the PLC.  Pylogix only reads/writes tags in the PLC, it cannot interact with the PC/PanelView.

**Reply by akashkumarmandal** [reply]
Thanks @dmroeder

**Reply by TheFern2** [reply]
If you have a SQL Server setup on your network and you have the creds you can query alarms with python with sql queries either raw psycopg2 or sqlalchemy if you feel fancy, your hmi's should sync up with it but each should have one as well basically if your hmi has alarm history, then you have a sql server setup locally or network. If you check on your sql server and have an alarms db with data in dbo.ConditionEvent then you have logged some alarms. You can also check with FT Admin Console and check that you have an alarms area otherwise it was never setup.

I think the table might be slightly diff in newer FT versions, but should give you an idea.
```
select * from dbo.ConditionEvent
where Message like '%keywork%';
```

```
where Message='Exact alarm message';
```

```
where EventTimeStamp like '%2020%';
```

Combined queries:
```
where (Message='DW_MC_F007 : DWA VFD Warning' and EventTimeStamp like '%2020%');
```

**Reply by evaldes2015** [reply]
If you have worked with other Python DBAPI packages you could also use python-tds to talk to MSSQL server.
